src/SUFST/src/can_unpack.c

```diff
@@ -42,7 +44,34 @@ unpack_ptr->rx_queue = can_queue;
                                TX_NO_TIME_SLICE,
                                TX_AUTO_START);
     }
+    if (tx_status == TX_SUCCESS)
+    {
+        tx_status = tx_queue_create(&unpack_ptr->rx_queue,
+                    "My Rx Queue",
+                    TX_1_ULONG,
+                    unpack_ptr->rx_queue_mem,
+                    sizeof(unpack_ptr->rx_queue));        
+    }
+    
+    // subscribe to a message
+    if (tx_status == TX_SUCCESS)
+    {
+        for(int i = 0; i < TABLE_SIZE; i++)
+        {
+            tx_status = rtcan_subscribe(&unpack_ptr->rtcan, can_handler_get(i)->identifier , &unpack_ptr->rx_queue);
```

Comment:
The type of `tx_status` is not correct here. RTCAN defines `rtcan_status_t` as a return type for all its functions (this is similar to `HAL_StatusTypeDef`). A new variable with type `rtcan_status_t` should be created to store this status code.

Reply:
The check for `RTCAN_OK` is still correct though.